### PR TITLE
GetColFamilies from the DB and getKeys for the colFamilies API

### PR DESCRIPTION
**Pull request by SriganeshNk**
Review and let me know what I'm missing here.

### CODE REVIEW

File: src/main/java/com/xjeffrose/chicago/client/BaseChicagoClient.java [hunk 1]
```diff
@@ -55,14 +55,17 @@ public void nodeRemoved() {
...
+    // this is only for test purpose. The zkServer should run on the same machine as that of the chicago server.
+    this.zkClient = new ZkClient(address.split(":")[0]+":2181", false);
+    this.zkClient.start();
     this.quorum = 1;
     ArrayList<String> nodeList = new ArrayList<>();
     nodeList.add(address);
     this.rendezvousHash =  new RendezvousHash(Funnels.stringFunnel(Charset.defaultCharset()), nodeList, quorum);
-    clientNodeWatcher = null;
+    // only if the zkserver runs in the same machine as the chicago server [Test purpose]
+    clientNodeWatcher = new ClientNodeWatcher(zkClient, rendezvousHash, listener);
```
Comment (SriganeshNk) [inline src/main/java/com/xjeffrose/chicago/client/BaseChicagoClient.java:68]: I assumed that this constructor would be more or less used for development with ZK running on the same machine as that of the "address" variable. Let me know if this is ok. I'll change this back. But this was needed for my local testing purpose.

File: src/main/java/com/xjeffrose/chicago/client/BaseChicagoClient.java [hunk 2]
```diff
@@ -55,14 +55,17 @@ public void nodeRemoved() {
   protected EventLoopGroup evg;
 
 
-  public BaseChicagoClient(String address){
+  public BaseChicagoClient(String address) throws InterruptedException{
     this.single_server = true;
-    this.zkClient = null;
+    // this is only for test purpose. The zkServer should run on the same machine as that of the chicago server.
+    this.zkClient = new ZkClient(address.split(":")[0]+":2181", false);
```
Comment (SriganeshNk) [inline src/main/java/com/xjeffrose/chicago/client/BaseChicagoClient.java:61]: Same as below

File: src/main/java/com/xjeffrose/chicago/ChiUtil.java [hunk 3]
```diff
@@ -13,6 +13,8 @@
   private static final Logger log = LoggerFactory.getLogger(ChiUtil.class.getName());
   public static final String delimiter = "@@@";
   public static final String defaultColFam = "chicago";
+  public static final String TS = "timeSeries";
+  public static final String KV = "KV";
```
Comment (SriganeshNk) [inline src/main/java/com/xjeffrose/chicago/ChiUtil.java:17]: The data that is written along with the path in ZK. Lets me know whether it is TimeSeries, in which case I could just stream from the beginning else I can scan the column family for the candidate keys.

**Comment by SriganeshNk** [comment]
Made some changes based on comments from Smarth. Please review and comment/merge.

File: src/main/java/com/xjeffrose/chicago/client/ChicagoClient.java [hunk 4]
```diff
@@ -67,6 +67,84 @@ public ChicagoClient(String address) throws InterruptedException {
...
+              e.printStackTrace();
+            } catch (TimeoutException e) {
+              e.printStackTrace();
+            }
+            if (ch1.isWritable()) {
+              UUID id1 = UUID.randomUUID();
+              SettableFuture<byte[]> f1 = SettableFuture.create();
+              Futures.addCallback(f1, new FutureCallback<byte[]>() {
+                @Override
+                public void onSuccess(@Nullable byte[] bytes) {
```
Comment (xjdr) [inline src/main/java/com/xjeffrose/chicago/client/ChicagoClient.java:124]: you aren't capturing the return value here. That could come back to byte you

**Pull request merged**